pkg/execution/state/redis_state/redis_state.go

```diff
@@ -757,7 +823,7 @@ type runMetadata struct {
 func (r runMetadata) Map() map[string]any {
 	return map[string]any{
 		"version":       r.Version,
-		"status":        r.Status,
+		"status":        int(r.Status), // Always store this as an int
```

Comment:
I'm guessing with these enums, then, that we should always add new values to the end of the list, else we risk skewing the data? 😬

Reply:
Yep, definitely.  New enums go at the end otherwise the values for those enums change.